mypy/semanal.py

```diff
@@ -671,6 +671,7 @@ def analyze_class_body(self, defn: ClassDef) -> Iterator[bool]:
             if self.analyze_typeddict_classdef(defn):
                 yield False
                 return
+            self.setup_class_def_analysis(defn)
```

Comment:
While this works, I'm not sure if this is the best change that could be made. As we can see in [`analyze_namedtuple_classdef`](https://github.com/elliott-beach/mypy/blob/ea573e23e08350783a4d3f8f98ee89cfdbb57983/mypy/semanal.py#L967):
(referencing [this TODO](https://github.com/elliott-beach/mypy/blob/ea573e23e08350783a4d3f8f98ee89cfdbb57983/mypy/semanal.py#L2287)), 
the root of the issue is that the analysis and processing of namedtuples is unconditionally global. If I understand things correctly, your change adds the node early to the symbol table so it is coerced into being a local definition by `setup_class_def_analysis`. That works, but is rather brittle.

Reply:
I agree that this is incomplete because we need to handle local definitions of namedutple in a few other places, but I would argue that we should call `setup_class_def_analysis` for namedtuples.  Local classes are added to the symbol table in `setup_class_def_analysis`, and there is no reason not to add local namedtuples at that point as well. 

This does not seem early to me because semantic analysis proceeds in two steps of (1) adding nodes with basic info to the symbol table and then (2) adding more complex type info. I think here is the right place to do step (1) for NamedTuples.

Reply:
The TODO comments by @JukkaL  appear to be referring to assigning `node.kind` as `GDEF` or `LDEF`.
If that is all, I can add commits to handle that, and then I think this would be complete.

Reply:
Okay I think this is acceptable.

Reply:
Great, Let me fix the todos.

Reply:
Actually, my understanding is LDEF and GDEF for NamedTuples, TypedDicts, and Enums is a longstanding issue that is not the cause of the specific problem this PR is solving, and I don't know how to devise a correct solution with confidence. So I won't address those.

Reply:
@JukkaL Would you look at this? I think you could merge this / request changes with confidence.